### ISSUE TITLE
Parameters for a TypedDict don't show type on hover

### ISSUE THREAD

**Issue by rchiodo** [microsoft/pyright#4124]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**
Hover for TypedDicts init does not compute the type

**To Reproduce**
Steps to reproduce the behavior.

![image](https://user-images.githubusercontent.com/19672699/199628230-a6c96e32-7c14-4dcf-af59-1e8d975fdb8c.png)

**VS Code extension or command-line**
VS Code Extension

**Additional context**

**Comment by erictraut** [comment]
@rchiodo, can you please past the code in text form? It's inconvenient to retype from a screen shot.

**Comment by tuchandra** [comment]
I don't think your code is correct. TypedDict is not initialized like a class, even though it looks like one. The below should work and autocomplete/hover correctly—

```python
y: Movie2 = {"name": "foo", "year": 4}
```

**Comment by erictraut** [comment]
@tuchandra, the constructor call in the screen shot above is legal for a TypedDict. The two forms you provided are also supported.

**Comment by erictraut** [comment]
Here's the code in text form:

```python
from typing import TypedDict

class Movie2(TypedDict):
    name: str
    year: int

y = Movie2(name='foo', year=4)

print(y['name'])
```

**Comment by erictraut** [comment]
I agree that the hover text in this case could be improved. Since this is a language server issue, not a core type-checking issue, I'm going to transfer it to pylance-release.
